### PR TITLE
Hide underlying debug adapter from the user

### PR DESCRIPTION
**Pull request by matthewbastien**
Users should not have to know which debug adapter is being used (CodeLLDB vs `lldb-dap`). The Swift extension will now always create a `swift-lldb` launch config for each executable that it finds in the workspace. When launched these configurations will either delegate to CodeLLDB or launch `lldb-dap` depending on which Swift toolchain version is currently being used. I've added a setting to force the usage of CodeLLDB when desired.

Issue: #1013 

Additionally, the `program` path in the launch configuration will always use POSIX style paths regardless of the platform. This allows the launch.json to be shared between users of different platforms.

Issue: #958

### CODE REVIEW

**Comment by adam-fowler** [comment]
Does this still give the user the option to use CodeLLDB when running Swift 6. 
What are the test plans for lldb-dap vs CodeLLDB?

**Comment by matthewbastien** [comment]
> Does this still give the user the option to use CodeLLDB when running Swift 6.

Yup! I renamed the old setting that governed this to `swift.debugger.useCodeLLDB` which, when enabled, will force the extension to always use CodeLLDB regardless of the Swift toolchain version. The behaviour is largely the same as it used to be, but the generated launch config will always be of the type `swift-lldb`. This has the added benefit that we no longer need to watch for settings changes to update the generated launch configs.

> What are the test plans for lldb-dap vs CodeLLDB?

Thanks for reminding me: I'll create a new issue to add tests for the debug adapter. A simple UI test to ensure that the debugger launches and hits breakpoints for each supported swift version should suffice (I'm not sure there's a good integration test for this and anything more than a simple UI test would be overkill and brittle in my experience). I'd also like to add unit tests for the creation of the launch config, but mocking the WorkspaceContext/FolderContext is too difficult at the moment. There'll probably need to be some changes to how the code is structured (potentially by adding DI) to make this easier. The overall testing plan for the extension right now is to add builds for macOS and Windows to cover all platforms.

**Comment by award999** [comment]
> I'm not sure there's a good integration test for this and anything more than a simple UI test would be overkill and brittle in my experience

https://www.npmjs.com/package/@vscode/debugadapter-testsupport may be an option if we connected on a port instead of starting it up https://github.com/microsoft/vscode-debugadapter-node/blob/main/testSupport/src/debugClient.ts#L88. IDK how feasible though

File: src/debugger/debugAdapter.ts [hunk 1]
```diff
@@ -24,18 +24,18 @@ import { SwiftToolchain } from "../toolchain/toolchain";
...
-            : "lldb";
+        return "swift-lldb";
     }
 
     /** Return debug adapter for toolchain */
-    public static getDebugAdapter(swiftVersion: Version): "lldb-vscode" | "lldb-dap" {
-        return swiftVersion.isLessThan(new Version(6, 0, 0)) ? "lldb-vscode" : "lldb-dap";
+    public static getDebugAdapterType(swiftVersion: Version): "lldb-vscode" | "lldb-dap" {
+        return configuration.debugger.useCodeLLDB
+            ? "lldb-vscode"
```
Comment (award999) [inline src/debugger/debugAdapter.ts:35]: question: does CodeLLDB even allow specifying a debug adapter?

File: package.json [hunk 2]
```diff
@@ -1276,4 +1276,4 @@
     "vscode-languageclient": "^9.0.1",
     "xml2js": "^0.6.2"
   }
-}
```
Comment (award999) [inline package.json:1279]: nit: each PR keeps changing this. Think there is an NPMRC setting to set consistent line ending

File: src/debugger/debugAdapter.ts (hunk 1 above)
Comment (award999) [reply src/debugger/debugAdapter.ts:35]: suggestion: assuming it is to make other code work below then maybe we should just adjust that logic

**Review by award999 — COMMENTED** [review]
looking great

File: package.json [hunk 3]
```diff
@@ -550,10 +550,10 @@
       {
         "title": "Debugger",
         "properties": {
-          "swift.debugger.useDebugAdapterFromToolchain": {
```
[the remark this answers is not in the document]
Comment (award999) [reply package.json:553]: I was thinking the same thing too but this only applies to Swift 6 right? So maybe not widely used. I guess we could use in `configuration.debugger.useCodeLLDB` as a fallback

File: src/debugger/debugAdapter.ts (hunk 1 above)
Comment (matthewbastien) [reply src/debugger/debugAdapter.ts:35]: No, CodeLLDB does not allow specifying a debug adapter. This is used in other places to determine which debug adapter we're using.

File: package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (matthewbastien) [reply package.json:553]: I've reverted this back to use the old setting so that we don't have to deprecate anything.

File: package.json (hunk 2 above)
Comment (matthewbastien) [reply package.json:1279]: Will create a separate pull request to enforce this behaviour.

**Comment by plemarquand** [comment]
@swift-server-bot test this please

**Comment by matthewbastien** [comment]
I've reverted back to using CodeLLDB by default for this patch until we can verify that lldb-dap has feature parity with CodeLLDB.

**Pull request merged**